### PR TITLE
[docs] Added Training models with GPU support section

### PR DESCRIPTION
**Pull request by gradientsky**
*Description of changes:*
Added Training models with GPU support section

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1527-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1527/1/index.html

File: docs/tutorials/tabular_prediction/tabular-gpu.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+# Training models with GPU support
+:label:`sec_tabulargpu`
+
+Training with GPU can significantly speed up base algorithms, but it is also a necessity for text and vision models where even on small tasks and the best CPUs the training can take forever without GPU acceleration. To enable GPU training, pass `num_gpus` parameter into a `fit()` call. All models
```
Comment (Innixma) [inline docs/tutorials/tabular_prediction/tabular-gpu.md:4]: Grammar: Training with GPU can significantly speed up base algorithms, and is a necessity for text and vision models where training without GPU is infeasibly slow.

File: docs/tutorials/tabular_prediction/tabular-gpu.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+with GPU support will automatically start using the acceleration.
+
+```{.python}
+save_path = 'agModels-predictClass'
```
Comment (Innixma) [inline docs/tutorials/tabular_prediction/tabular-gpu.md:8]: Why save_path? This is unrelated to the tutorial

File: docs/tutorials/tabular_prediction/tabular-gpu.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+predictor = TabularPredictor(label=label, path=save_path).fit(
+    train_data,
+    ag_args_fit={'num_gpus': 1}
+)
+```
+
+If GPU acceleration should be enabled just on a specific models, the same parameter can be passed into model `hyperparameters`:
+
+```{.python}
+save_path = 'agModels-predictClass'  # specifies folder to store trained models
```
Comment (Innixma) [inline docs/tutorials/tabular_prediction/tabular-gpu.md:18]: Why save_path?

File: docs/tutorials/tabular_prediction/tabular-gpu.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+}
+predictor = TabularPredictor(label=label, path=save_path).fit(
+    train_data, 
+    hyperparameters=hyperparameters, 
+)
+```
+
+## Multi-modal
+
+In :ref:`sec_tabularprediction_multimodal` tutorial we presented how to train an ensemble which can utilize tabular, text and images. If available GPU don't have enough RAM to handle default model sizes or it is needed to speedup testing, different backends can be used:
```
Comment (Innixma) [inline docs/tutorials/tabular_prediction/tabular-gpu.md:33]: GPU -> GPUs, RAM -> VRAM, different backends can be used: -> different backends can be used.

File: docs/tutorials/tabular_prediction/tabular-gpu.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+### Vision models
+
+Text model preset to use can be set via:
+
+```{.python}
+hyperparameters['AG_IMAGE_NN'] = {'model': '<model>'}
+```
+
+The list of available text model presets is:
```
Comment (Innixma) [inline docs/tutorials/tabular_prediction/tabular-gpu.md:66]: Available text model presets:

File: docs/tutorials/tabular_prediction/tabular-gpu.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
+# Training models with GPU support
```
Comment (Innixma) [inline docs/tutorials/tabular_prediction/tabular-gpu.md:1]: No mention of how to install CUDA / check that CUDA is available / check that AutoGluon detects the GPU?

File: docs/tutorials/tabular_prediction/tabular-gpu.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+The list of available text model presets is:
+
+```{.python .input}
+from autogluon.vision.predictor.predictor import _get_supported_models
+_get_supported_models()[:10]  # there're more, we just show a few
+```
+
+## Enabling GPU for LightGBM
+
+When installed AutoGluon, out-of-the box it is using the LightGBM from a default installation. If `num_gpus` is set, the following warning will be displayed:
```
Comment (Innixma) [inline docs/tutorials/tabular_prediction/tabular-gpu.md:75]: When installed AutoGluon, out-of-the box it is using the LightGBM from a default installation. -> The default installation of LightGBM does not support GPU training, however GPU support can be enabled via a special install.

File: docs/tutorials/tabular_prediction/tabular-gpu.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+```{.python}
+save_path = 'agModels-predictClass'
+predictor = TabularPredictor(label=label, path=save_path).fit(
+    train_data,
+    ag_args_fit={'num_gpus': 1}
+)
+```
+
+If GPU acceleration should be enabled just on a specific models, the same parameter can be passed into model `hyperparameters`:
```
[the remark this answers is not in the document]
Comment (gradientsky) [reply docs/tutorials/tabular_prediction/tabular-gpu.md:15]: Fixed

File: docs/tutorials/tabular_prediction/tabular-gpu.md (hunk 1 above)
Comment (gradientsky) [reply docs/tutorials/tabular_prediction/tabular-gpu.md:4]: fixed

File: docs/tutorials/tabular_prediction/tabular-gpu.md (hunk 2 above)
Comment (gradientsky) [reply docs/tutorials/tabular_prediction/tabular-gpu.md:8]: removed

File: docs/tutorials/tabular_prediction/tabular-gpu.md (hunk 3 above)
Comment (gradientsky) [reply docs/tutorials/tabular_prediction/tabular-gpu.md:18]: removed

File: docs/tutorials/tabular_prediction/tabular-gpu.md (hunk 5 above)
Comment (gradientsky) [reply docs/tutorials/tabular_prediction/tabular-gpu.md:66]: fixed

File: docs/tutorials/tabular_prediction/tabular-gpu.md (hunk 6 above)
Comment (gradientsky) [reply docs/tutorials/tabular_prediction/tabular-gpu.md:1]: We should not bring CUDA installation topic into our docs: it's changing over time and across operation systems. I added a note that GPU traning requires CUDA installed and the official documentation [1] should be followed.

[1] https://docs.nvidia.com/cuda/

File: docs/tutorials/tabular_prediction/tabular-gpu.md (hunk 7 above)
Comment (gradientsky) [reply docs/tutorials/tabular_prediction/tabular-gpu.md:75]: updated

File: docs/tutorials/tabular_prediction/tabular-gpu.md (hunk 4 above)
Comment (gradientsky) [reply docs/tutorials/tabular_prediction/tabular-gpu.md:33]: updated

**Review by Innixma — APPROVED** [review]
LGTM!
